core/offchain/impl/runner.hpp

```diff
@@ -22,12 +25,18 @@ namespace kagome::offchain {
    public:
     using Task = std::function<void()>;
 
-    Runner(size_t threads, size_t max_tasks)
+    Runner(size_t threads = kMaxThreads, size_t max_tasks = kMaxTasks)
         : threads_{threads},
           free_threads_{threads},
           max_tasks_{max_tasks},
           thread_pool_{"ocw", threads_} {}
 
+    Runner(std::nullptr_t,
+           std::nullptr_t,
+           std::nullptr_t,
+           std::nullptr_t)  // Hack Boost.DI
+        : Runner(kMaxThreads, kMaxTasks) {}
```

Comment:
di selects longest constructor,
`...` consumes everything,
explicit `Inject{}` to avoid implicit `{}`
```suggestion
    struct Inject { explicit Inject() = default; };
    Runner(Inject, ...) : Runner{} {}
```

(or `BOOST_DI_INJECT` from `<boost/di/inject.hpp>`)